Tests/NIOCoreTests/AsyncSequences/NIOAsyncSequenceTests.swift

```diff
@@ -288,6 +288,10 @@ final class NIOAsyncSequenceProducerTests: XCTestCase {
     func testFinish_whenFinished() async throws {
         self.source.finish()
 
+        XCTAssertEqual(self.delegate.didTerminateCallCount, 0)
+
+        _ = await self.sequence.first { _ in true }
+
```

Comment:
@Lukasa @glbrntt This is a slight change in behaviour where I am now calling `didTerminate` only after the `nil` or thrown `Error` has been consumed. I was going back and forth about this but thought this was the best way. Before we already called `didTerminate` once the last element has been consumed. WDYT?

Reply:
I think this is fine.